src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+import PushNotification from 'react-native-push-notification';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
+import messaging from '@react-native-firebase/messaging';
+import React, {useState} from 'react';
+import {Platform} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
+import {useRecoilState} from 'recoil';
+
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import profileAtom from '../../../atom/profileAtom';
+import useChatUtilsHook from '../chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import TokenStorage, {ITokenEnum} from '../../../utils/storage/custom/tokenStorage';
+import {getAnonymousUserId, getUserId} from '../../../utils/users';
+
+const usePushNotificationHook = () => {
+  const isIos = Platform.OS === 'ios';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `usePushNotificationHook` function is using a lot of hooks and states. This could lead to unnecessary re-renders and performance issues. Consider refactoring the code to minimize the number of hooks and states used.



<!-- This is an auto-generated comment by OSS CodeRabbit -->